tests/integration/deploy_txn_tests.rs

```diff
@@ -0,0 +1,110 @@
+use blockifier::context::BlockContext;
+use blockifier::deploy_account_tx_args;
+use blockifier::test_utils::deploy_account::deploy_account_tx;
+use blockifier::test_utils::{NonceManager, BALANCE};
+use blockifier::transaction::account_transaction::AccountTransaction;
+use blockifier::transaction::test_utils::max_fee;
+use rstest::{fixture, rstest};
+use starknet_api::core::{calculate_contract_address, ClassHash, ContractAddress, PatriciaKey};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_api::transaction::{Calldata, ContractAddressSalt, Fee, TransactionVersion};
+use starknet_api::{class_hash, contract_address, patricia_key, stark_felt};
+
+use crate::common::block_context;
+use crate::common::state::{init_logging, load_cairo0_contract, StarknetStateBuilder, StarknetTestState};
+use crate::common::transaction_utils::execute_txs_and_run_os;
+
+#[derive(Debug)]
+struct DeployArgs {
+    contract_address_salt: ContractAddressSalt,
+    class_hash: ClassHash,
+    constructor_calldata: Vec<StarkFelt>,
+    deployer_address: ContractAddress,
+}
+
+/// Fixture to create an initial Starknet state for deploy tx tests.
+///
+/// One specificity of deploy txs is that the deployed contract funds its own deployment.
+/// This means that the state must already provision balances for a yet-to-be-deployed contract.
+/// For these tests we compute the contract address in advance and set funds accordingly.
+#[fixture]
+pub async fn initial_state_for_deploy_v1(
+    block_context: BlockContext,
+    #[from(init_logging)] _logging: (),
+) -> (StarknetTestState, DeployArgs) {
+    let account_with_dummy_validate = load_cairo0_contract("account_with_dummy_validate");
+    let account_with_long_validate = load_cairo0_contract("account_with_long_validate");
+
+    // This is the hardcoded class hash of `account_with_long_validate`.
+    // Recomputing it automatically requires a significant amount of code reorganization so
+    // we hardcode it for simplicity.
+    let class_hash = class_hash!("0x067605bc345e925118dd60e09888a600e338047aa61e66361d48604ea670b709");
```

Comment:
Hmmm, could you call `to_internal_tx()` (or `to_internal_deploy_v1()`) fisrt to convert the tx and then easily call `tx_hash_deploy_account_v1()`?

Reply:
Related: https://xkcd.com/1205/

Reply:
It's the class hash, not the tx hash so no can do. The only method we have for that is to use the Pathfinder hash functions, and I don't want to bother with the type conversion here (or set up a storage + FFC and write a compiled class fact to it).